src/sage/schemes/elliptic_curves/ell_field.py

```diff
@@ -1113,6 +1113,24 @@ def division_field(self, n, names='t', map=False, **kwds):
             L = L, F_to_K.post_compose(K_to_L)
         return L
 
+    def _Hom_(*args, **kwds):
+        r"""
+        Hook to make :class:`~sage.categories.homset.Hom`
+        set the correct parent
+        :class:`~sage.schemes.elliptic_curves.homset.EllipticCurveHomset`
+        for
+        :class:`~sage.schemes.elliptic_curves.hom.EllipticCurveHom`
+        objects.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(19), [1,0])
+            sage: type(E._Hom_(E))
+            <class 'sage.schemes.elliptic_curves.homset.EllipticCurveHomset_with_category'>
+        """
+        from . import homset
+        return homset.EllipticCurveHomset(*args, **kwds)
```

Comment:
You should have some more restrictions on the input as the `_Hom_` method is called very generically IIRC. Try, e.g., `Hom(E, R['x','y'])`. I would be more specific about the inputs allowed too.